specification/walkthrough.md

```diff
@@ -0,0 +1,273 @@
+trials.
+
+A basic analysis of this task is to determine which regions showed _greater
+activity for incongruent versus congruent trials_, across all participants.
+
+We can perform this analysis by first estimating a **run-level** timeseries
+model for "Incongruent" and "Congruent" trials--separately for each individual
+run. We then compute a contrast comparing _Incongruent > Congruent_ (IvC)
+trials. Next, we pass the resulting statistical maps for the contrast to a
+**subject-level** estimator, which computes the average _IvC_ effect for each
+subject separately. Finally, we pass the resulting estimates to a
+**dataset-level** estimator, which conduts a one-sample t-test across all of the
+subject estimates for the _IvC_ contrast.
+
+Let's visualize this model for 3 participants:
+
+![flow](images/flow_model.png)
+
+We can formally represent this analysis as **BIDS Stats Model**:
+
+```{literalinclude} examples/model-walkthrough_smdl.json
+:language: JSON
+```
+
+```{note}
+For this example, we have limited the model to three subjects using the `Input` key.
+```
+
+_BSM_ defines this multi-stage analysis as a Graph, with each level of analysis
+defined as a separate `Node` object.
+
+### Run-level Model
+
+First, we define a `Node` for the run level analysis.
+
+```json
+      "Level": "Run",
+      "Name": "run_level"
+```
+
+Note that the {py:attr}`~bsmschema.models.Node.Level` key is necessary for
+determining which input images are available to the estimator. At the `Run`
+level, there are many sources of possible variables, most notably `_events.tsv`
+files which define the timing of task-related events.
+
+Next we define a {py:attr}`~bsmschema.models.Model` for this node.
+
+```json
+      "Model": {
+        "X": [1, "incongruent", "congruent"],
+        "Type": "glm"
+      }
+```
+
+The {py:attr}`~bsmschema.models.Model.X` parameter defines the variables in the
+design matrix. Here, we are modeling the `incongruent` and `congruent` trial
+types, in addition to an intercept (idenitified by the special key: `1`; see:
+\_).
+
+Next, we specify an _Incongruent-Congruent (IvC)_ contrast using the
+{py:attr}`~bsmschema.models.Contrast` key:
+
+```json
+      "Contrasts": [
+        {
+          "Name": "IvC",
+          "ConditionList": ["incongruent", "congruent"],
+          "Weights": [1, -1],
+          "Test": "t"
+        }
+      ]
+```
+
+If you have used other fMRI modeling tools this should be familar. We have
+specified a t-test contrast with the weights `[1, -1]` for the conditions:
+`["incongruent", "congruent"]` and given this contrast the name `IvC`.
+
+```{attention}
+`Contrasts` **define the outputs** that will be available to the next `Node`.
+
+Since we only modeled a single contrast (`IvC`), the next `Node` will not have access to estimates for main effects for the `congruent` or `incongruent` conditions, unless we explicitly compute a `Contrast` for each.
+```
+
```

Comment:
```suggestion

By now our whole BSM should looks like this:

```json
{
  "Name": "Simon IvC",
  "BIDSModelVersion": "1.0.0",
  "Input": { "subject": ["01", "02", "03"] },
  "Nodes": [
    {
      "Level": "Run",
      "Name": "run_level",
      "GroupBy": ["run", "subject"],
      "Model": { "X": [1, "incongruent", "congruent"], "Type": "glm" },
      "Contrasts": [
        {
          "Name": "IvC",
          "ConditionList": ["incongruent", "congruent"],
          "Weights": [1, -1],
          "Test": "t"
        }
      ]
    }
  ]
}
```

```

Reply:
I rather not build it up sequentially. Instead I'm going to refer to parts of the complete model. Otherwise we risk having an incomplete and non-functional example.

Reply:
I am ok having incomplete models at certain steps in the tuto but I won't fight you on this.

Reply:
The only thing I would be willing to do (immediately followed by a complete example--unlike the first time this was an issue), would be a skeleton model showing the structure (i.e. `Nodes`, `Edges` and `Input`). I'm going to play around with it and see what people think. Maybe with highlighting it could work.

I do think a gentle start is worthwhile.